freqtrade/optimize/hyperopt.py

```diff
@@ -557,6 +558,8 @@ def start(self) -> None:
         cpus = cpu_count()
         logger.info(f"Found {cpus} CPU cores. Let's make them scream!")
         config_jobs = self.config.get('hyperopt_jobs', -1)
+        if self.total_epochs < cpus and (config_jobs > self.total_epochs or config_jobs < 0):
+            config_jobs = self.total_epochs
```

Comment:
Small issue with this:
assume we have 8 cpus, total_epochs=7 and config_jobs=-3. This setting of config_jobs assumes, that the number of workers will be set to 6 (8+1-3), leaving 2 cpus free for the system. With your code this will be set to 7.
Same for 4 cpus, total_epochs=3 and also config_jobs=-3. config_jobs should be set to 2 (or left as -3).

I propose not changing negative values of config_jobs (but this should be properly handles below for the last loop iteration)...

Reply:
Fixed the logic to only change in case the value is not OK also for the scenario you described:
```
        if (config_jobs < 0 and (cpus + config_jobs + 1) > self.total_epochs) \
           or (config_jobs > 0 and config_jobs > self.total_epochs):
            config_jobs = self.total_epochs
```